include/nanobind/stl/filesystem.h

```diff
@@ -0,0 +1,105 @@
+/*
+    nanobind/stl/filesystem.h: type caster for std::string
+
+    Copyright (c) 2022 Qingnan Zhou and Wenzel Jakob
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+#pragma once
+
+#include <nanobind/nanobind.h>
+#include <string>
+
+#ifdef __has_include
+#if __has_include(<filesystem>)
+#include <filesystem>
+#define NB_HAS_FILESYSTEM 1
+#elif __has_include(<experimental/filesystem>)
+#include <experimental/filesystem>
+#define NB_HAS_EXPERIMENTAL_FILESYSTEM 1
+#endif
+#endif
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+template <typename T>
+struct path_caster {
+private:
+    static PyObject *unicode_from_fs_native(const std::string &w) {
```

Comment:
Can you update these functions so that they return a `str` object? You will have to wrap the result into `steal<str>(result)`. Then that detail can be removed from functions calling `unicode_from_fs_native` (it actually seems unsafe how these functions are being used right now with respect to exceptions)

Reply:
I have pushed a change.  Let me know if I understand your request correctly on this one.